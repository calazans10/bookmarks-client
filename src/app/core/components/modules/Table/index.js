import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './styles';

const Table = ({ children }) => <Container>{children}</Container>;

Table.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Table;
