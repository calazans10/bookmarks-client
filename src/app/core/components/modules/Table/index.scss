@import 'styles/utilities';

.table {
  width: 100%;
  border-right: rem(2) solid #efefef;
  border-left: rem(2) solid #efefef;
  border-collapse: collapse;
  border-spacing: 0;

  @media(min-width: 600px) {
    margin: 0 auto rem(40);
  }
}
