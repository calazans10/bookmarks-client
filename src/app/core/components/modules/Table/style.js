import styled from 'styled-components';
import { rem } from 'polished';

export const Container = styled.table`
  width: 100%;
  margin-bottom: ${rem('40px')};
  border-right: ${rem('2px')} solid #efefef;
  border-left: ${rem('2px')} solid #efefef;
  border-collapse: collapse;
  border-spacing: 0;

  @media (min-width: 600px) {
    margin: 0 auto ${rem('40px')};
  }
`;
