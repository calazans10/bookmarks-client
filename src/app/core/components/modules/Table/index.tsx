import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './style';

const Table = ({ children }) => <Container>{children}</Container>;

Table.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Table;
