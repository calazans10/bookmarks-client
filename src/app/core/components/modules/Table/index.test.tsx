import React from 'react';
import { render } from '@testing-library/react';
import Table from '.';

describe('Table', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <Table>
        <tbody>
          <tr>
            <th>Title</th>
          </tr>
          <tr>
            <td>Test</td>
          </tr>
        </tbody>
      </Table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
