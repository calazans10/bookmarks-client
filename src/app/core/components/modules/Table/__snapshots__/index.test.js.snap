// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders without crashing 1`] = `
<table
  class="sc-AykKC bBOGGK"
>
  <tbody>
    <tr>
      <th>
        Title
      </th>
    </tr>
    <tr>
      <td>
        Test
      </td>
    </tr>
  </tbody>
</table>
`;
